### PR TITLE
Serve hello world html page

### LINKED ISSUE #6: Hello world server
Delivers a server-rendered html page that says 'hello world'

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes #6.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,11 +1,98 @@
...
+            head {
+                title {
+                    "foo"
+                }
+            }
+            body {
+                h1 { "hello world" }
+            }
+        }
+    };
```
Comment (soenkehahn) [inline src/main.rs:42]: I've used `maud` for html rendering now. Not sure whether I like that though.

File: Cargo.toml [hunk 2]
```diff
@@ -5,3 +5,14 @@ authors = ["Casey Rodarmor <casey@rodarmor.com>", "Sönke Hahn <soenkehahn@gmail
 edition = "2018"
 
 [dependencies]
+maud = "0.22.2"
```
Comment (casey) [inline Cargo.toml:8]: How about an empty line after this line, just for cosmetic purposes?

**Comment by casey** [comment]
Nice, looks good!

The compile-time template library that I used before is [askama](https://github.com/djc/askama). It's more of a traditional text-template engine, with a `templates` directory full of template files, which are mostly text aside from `{{...}}` interpolations.

To use a template, you declare a struct like this:

```rust
#[derive(Template)]
#[template(path = "hello.html")]
struct Foo {
  bar: u32,
  baz: String,
}
```

And can then render it like so:

```rust
Foo { bar: 11, baz: "foo".into() }.render().unwrap()
```

I see maud is a bit more like JSX, with inline macros that generate HTML. I've never used it, but it and askama have a similar amount of stars, and the code looks good.  I kind of prefer the askama approach, and like the separation of template files and code. But, it isn't a strong preference. And since I haven't used maud, I won't mind trying it out and seeing how it feels. It shouldn't be too hard to switch later if we need to.

What do you think?

**Comment by casey** [comment]
(If you want to go with maud, feel free to merge as is.)

File: Cargo.toml (hunk 2 above)
Comment (soenkehahn) [reply Cargo.toml:8]: Good catch. This was `cargo add`'s doing.

**Pull request merged**